### PR TITLE
Added Version Installer and for Kubernetes Task

### PR DESCRIPTION
**Pull request by Anumita**

### CODE REVIEW

File: Tasks/Kubernetes/make.json [hunk 1]
```diff
@@ -10,5 +10,14 @@
 		"type": "node",
 		"dest" : "./",
 		"compile" : true
-	}]
+	}],
+	"rm": [
+        {
+            "items": [
+                "node_modules/utility-common/node_modules/vsts-task-lib",
+				"node_modules/docker-common/node_modules/vsts-task-lib"
```
Comment (jikuma) [inline Tasks/Kubernetes/make.json]: indentation. Create a separate task to investigate why is it required now?

File: Tasks/Kubernetes/package.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "dependencies": {
     "del": "2.2.0",
-    "js-yaml": "3.6.1"
+    "js-yaml": "3.6.1",
+	"vsts-task-tool-lib": "^0.4.0"
```
Comment (jikuma) [inline Tasks/Kubernetes/package.json]: Remove ^ and fix it to a version 0.4.0. When you add/change a dependency you have to also create a swagger file required for licencing. Create a task to do it if you want to complete the PR today.

File: Tasks/Kubernetes/task.json [hunk 3]
```diff
@@ -193,8 +192,8 @@
             "type": "filePath",
             "label": "Path to Kubectl",
```
Comment (jikuma) [inline Tasks/Kubernetes/task.json]: it's a file picker right?

File: Tasks/Kubernetes/src/clusterconnection.ts [hunk 4]
```diff
@@ -24,8 +24,21 @@ export default class ClusterConnection {
         {
             tl.debug(tl.loc("DownloadingClient"));
             this.kubectlPath = path.join(this.userDir, "kubectl") + this.getExecutableExtention();
-            var version = await utils.getStableKubectlVersion();
-            await utils.downloadKubectl(version, this.kubectlPath);
+            let versionOrLocation = tl.getInput("versionOrLocation");
+			if(versionOrLocation === "version")
```
Comment (jikuma) [inline Tasks/Kubernetes/src/clusterconnection.ts]: indentation.

File: Tasks/Kubernetes/src/utilities.ts [hunk 5]
```diff
@@ -34,6 +35,26 @@ function ensureDirExists(dirPath : string) : void
     }
 }
 
+export async function getKubectlVersion(versionSpec: string, checkLatest: boolean, kubectlPath: string) : Promise<string> {
```
Comment (jikuma) [inline Tasks/Kubernetes/src/utilities.ts]: is kubectlPath input required for this function?

File: Tasks/Kubernetes/src/utilities.ts [hunk 6]
```diff
@@ -34,6 +35,26 @@ function ensureDirExists(dirPath : string) : void
...
+   let version: string;   
+   let versionPrefex = 'v';
+   let explicitVersion = false;
+   if (versionSpec && toolLib.isExplicitVersion(versionSpec)) {
+	    version = versionPrefex.concat(versionSpec);
+        checkLatest = false; // check latest doesn't make sense when explicit version
+		explicitVersion = true;
```
Comment (jikuma) [inline Tasks/Kubernetes/src/utilities.ts]: indentation.

File: Tasks/Kubernetes/src/clusterconnection.ts [hunk 7]
```diff
@@ -24,8 +24,21 @@ export default class ClusterConnection {
...
+			{
+				let versionSpec = tl.getInput("versionSpec");
+				let checkLatest: boolean = tl.getBoolInput('checkLatest', false);
+				let version  = await utils.getKubectlVersion(versionSpec, checkLatest, this.kubectlPath);
+				await utils.downloadKubectl(version, this.kubectlPath);
+			}
+			else if( versionOrLocation === "location")
+			{
+				let pathToKubectl = tl.getPathInput("specifyLocation", false, true);
```
Comment (jikuma) [inline Tasks/Kubernetes/src/clusterconnection.ts]: When location is provided just check if the file exists and change the value of this.kubectlPath to specifyLocation and do chmod. This way we don't have to copy to a location. Secondly, it has a benefit that if kubectl is corrupt user will come to know in the log and will not blame us for using any other kubectl of our choice.

File: Tasks/Kubernetes/src/utilities.ts [hunk 8]
```diff
@@ -34,6 +35,26 @@ function ensureDirExists(dirPath : string) : void
...
+    }
+   
+   if (checkLatest || !versionSpec || !explicitVersion) {
+        version = await getStableKubectlVersion();
+        if (!version) {
+            throw new Error(`Unable to find Kubectl version '${versionSpec}' for platform ${os.type()}.`);
+        }
+   }
+	
+	return version;
```
Comment (jikuma) [inline Tasks/Kubernetes/src/utilities.ts]: indentation.

File: Tasks/Kubernetes/src/clusterconnection.ts [hunk 9]
```diff
@@ -24,8 +24,21 @@ export default class ClusterConnection {
         {
             tl.debug(tl.loc("DownloadingClient"));
             this.kubectlPath = path.join(this.userDir, "kubectl") + this.getExecutableExtention();
-            var version = await utils.getStableKubectlVersion();
-            await utils.downloadKubectl(version, this.kubectlPath);
+            let versionOrLocation = tl.getInput("versionOrLocation");
```
Comment (jikuma) [inline Tasks/Kubernetes/src/clusterconnection.ts]: Let's move this logic to a function GetKubectlPath which take the inputs and return kubectl path.

File: Tasks/Kubernetes/task.json (hunk 3 above)
Comment (Anumita) [reply Tasks/Kubernetes/task.json]: yes

**Pull request merged**